devbuild.sh

```diff
@@ -1,76 +1,235 @@
+    --components=) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --clean) CLEAN=true ;;
+    --clean=?*) printf "ERROR: $1 argument ignored.\n"; usage; exit 1 ;;
+    --clean=) printf "ERROR: $1 argument ignored.\n"; usage; exit 1 ;;
+    --continue) CONTINUE=true ;;
+    --continue=?*) printf "ERROR: $1 argument ignored.\n"; usage; exit 1 ;;
+    --continue=) printf "ERROR: $1 argument ignored.\n"; usage; exit 1 ;;
+    --build-dir=?*) BUILD_DIR=${1#*=} ;;
+    --build-dir) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --build-dir=) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --install-dir=?*) INSTALL_DIR=${1#*=} ;;
+    --install-dir) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --install-dir=) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --build-dir=?*) BUILD_DIR=${1#*=} ;;
+    --build-dir) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --build-dir=) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --build-type=?*) BUILD_TYPE=${1#*=} ;;
+    --build-type) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --build-type=) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --build-jobs=?*) BUILD_JOBS=$((${1#*=})) ;;
+    --build-jobs) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --build-jobs=) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
+    --verbose|-v) VERBOSE=true ;;
+    --verbose=?*) printf "ERROR: $1 argument ignored.\n"; usage; exit 1 ;;
+    --verbose=) printf "ERROR: $1 argument ignored.\n"; usage; exit 1 ;;
+    -?*) printf "ERROR: Unknown option $1\n"; usage; exit 1 ;;
+    *) break
+  esac
+  shift
+done
+
+set -eu
+
+# automatically determine system
+if [ -z "${PLATFORM}" ] ; then
+  PLATFORM=$(find_system)
 fi
-if [ $# -gt 2 ]; then
-  echo "ERROR: too many arguments"
-  usage
-  exit 1
+
+# automatically determine compiler
+if [ -z "${COMPILER}" ] ; then
+  case ${PLATFORM} in
+    jet|hera) COMPILER=intel ;;
+    orion) COMPILER=intel ;;
+    wcoss) COMPILER=cray_intel ;;
+    cheyenne) COMPILER=intel ;;
+    macos) COMPILER=gccgfortran ;;
+    *) printf "ERROR: Unknown platform $PLATFORM\n"; usage; exit 1 ;;
+  esac
 fi
 
-if [ "$1" = "--help" ] || [ "$1" = "-h" ]; then 
-  usage
-  exit 0
+# print settings
+if [ "${VERBOSE}" = true ] ; then
+  settings
 fi
 
-ENV_FILE="env/build_${PLATFORM}_${COMPILER}.env"
-if [ ! -f "$ENV_FILE" ]; then
-  echo "ERROR: environment file ($ENV_FILE) does not exist for this platform/compiler combination"
-  echo "PLATFORM=$PLATFORM"
-  echo "COMPILER=$COMPILER"
-  echo ""
-  echo "See User's Guide for detailed build instructions"
+# set ENV_FILE for this platform/compiler combination
+ENV_FILE="${SRC_DIR}/env/build_${PLATFORM}_${COMPILER}.env"
+if [ ! -f "${ENV_FILE}" ]; then
+  printf "ERROR: environment file does not exist for this platform/compiler\n"
```

Comment:
Combine `printf`s?

Reply:
@gsketefian I only combined printf statements for the large blocks in usage and settings. I like that the printf statements in ENV_FILE and BUILD_DIR are indented to follow the code. Is this okay?

Reply:
Yes, no problem.